crates/burn-autodiff/src/backend.rs

```diff
@@ -46,37 +52,42 @@ impl<B: Backend, C: CheckpointStrategy> Backend for Autodiff<B, C> {
     fn sync(device: &B::Device) {
         B::sync(device);
     }
+
+    fn dyn_from_data(
+        data: DynData<Self::FullPrecisionElem, Self::IntElem>,
+        device: &Self::Device,
+    ) -> Self::DynTensorPrimitive {
+        B::dyn_from_data(data, device)
+    }
+
+    fn dyn_into_data(
+        dyn_tensor: Self::DynTensorPrimitive,
+    ) -> DynData<Self::FullPrecisionElem, Self::IntElem> {
+        B::dyn_into_data(dyn_tensor)
+    }
 }
```

Comment:
Those functions can't exist in their current form, reading dynamic tensors that can be int, bool or float can't be encoded with FullPrecisionElement. Either we read them to bytes (u8) or we just don't support reading dynamic tensors.

Reply:
I'm not sure I understand this. Could you elaborate on your statement?